pvlib/irradiance.py

```diff
@@ -31,6 +30,90 @@
                    'sea': 0.06}
 
 
+def cloud_opacity_factor(irr_dif_clouds: np.ndarray,
+                         irr_dir_clouds: np.ndarray,
+                         irr_ghi_clouds: np.ndarray,
+                         spectra: dict) -> (np.ndarray, np.ndarray):
```

Comment:
Although we might add type hints eventually (see e.g. https://github.com/pvlib/pvlib-python/pull/1146#issuecomment-778386641), I think for now better to keep the signature unannotated (and see below points about types anyway).